observation_portal/requestgroups/serializers.py

```diff
@@ -25,10 +26,83 @@
 from datetime import timedelta
 from observation_portal.common.rise_set_utils import get_filtered_rise_set_intervals_by_site, get_largest_interval
 
-
 logger = logging.getLogger(__name__)
 
 
+class ModeValidator:
+    """Class used to validate modes of different types"""
+    def __init__(self, mode_type, instrument_type, default_modes, modes):
+        self._mode_type = mode_type.lower()
+        self._instrument_type = instrument_type
+        self._default_modes = default_modes
+        self._modes = modes
+        self._mode_key = 'rotator_mode' if self._mode_type == 'rotator' else 'mode'
```

Comment:
why is this different for rotator?

Reply:
Because in the configuration dict (which could be the instrument_config or guiding_config... whatever is passed in) the rotator mode is the set in the `rotator_mode` field, and all the other modes are just set in the `mode` field in their respective configuration dicts.